src/equations/ideal_glm_mhd_1d.jl

```diff
@@ -0,0 +1,715 @@
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
+end
+
+
+"""
+    initial_condition_ryujones_shock_tube(x, t, equations::IdealGlmMhdEquations1D)
+
+Ryu and Jones shock tube test case for one dimensional ideal MHD equations. Contains
+fast shocks, slow shocks, and rational disconinuties that propagate on either side
+of the contact discontinuity. Exercises the scheme to capture all 7 types of waves
+present in the one dimensional MHD equations. It is the second test from Section 4 of
+- Ryu and Jones (1995)
+  Numerical Magnetohydrodynamics in Astrophysics: Algorithm and Tests
+  for One-Dimensional Flow
+  [DOI: 10.1086/175437](https://doi.org/10.1086/175437)
+!!! note
+    This paper has a typo in the initial condtions. Their vairable `E` should be `p`.
+"""
+function initial_condition_ryujones_shock_tube(x, t, equations::IdealGlmMhdEquations1D)
+  # domain must be set to [0, 1], γ = 5/3, final time = 0.2
+  rho = x[1] <= 0.5 ? 1.08 : 1.0
+  v1 = x[1] <= 0.5 ? 1.2 : 0.0
+  v2 = x[1] <= 0.5 ? 0.01 : 0.0
+  v3 = x[1] <= 0.5 ? 0.5 : 0.0
+  p = x[1] <= 0.5 ? 0.95 : 1.0
+  inv_sqrt4pi = 1.0 / sqrt(4 * pi)
+  B1 = 2 * inv_sqrt4pi
+  B2 = x[1] <= 0.5 ? 3.6 * inv_sqrt4pi : 4.0 * inv_sqrt4pi
+  B3 = B1
+
+  return prim2cons(SVector(rho, v1, v2, v3, p, B1, B2, B3), equations)
+end
+
+"""
+    boundary_condition_ryujones_shock_tube(u_inner, orientation, direction, x, t,
+                                            surface_flux_function,
+                                            equations::IdealGlmMhdEquations1D)
+
+Boundary conditions used for the Ryu and Jones shock tube in combination with
+[`initial_condition_ryujones_shock_tube`](@ref).
+"""
+function boundary_condition_ryujones_shock_tube(u_inner, orientation, direction, x, t,
+                                                surface_flux_function,
+                                                equations::IdealGlmMhdEquations1D)
+  u_boundary = initial_condition_ryujones_shock_tube(x, t, equations)
+  # Calculate boundary flux
+  if direction == 2 # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
+end
+
+
+"""
+    initial_condition_shu_osher(x, t, equations::IdealGlmMhdEquations1D)
+
+Extended version of the test of Shu and Osher for one dimensional ideal MHD equations.
+Taken from Section 4.1 of
+- Derigs et al. (2016)
+  A Novel High-Order, Entropy Stable, 3D AMR MHD Solver withGuaranteed Positive Pressure
+  [DOI: 10.1016/j.jcp.2016.04.048](https://doi.org/10.1016/j.jcp.2016.04.048)
+"""
+function initial_condition_shu_osher(x, t, equations::IdealGlmMhdEquations1D)
```

Comment:
Should you decide to rename the files, you should also rename the initial/boundary conditions to `shuosher_shock_tube`